src/components/CheckIn/CheckIn.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useEffect, useState } from 'react'
+import { uuid4 } from '@sentry/utils'
+import { ConfirmDialog } from './internal/ConfirmDialog'
+import { DummyCheckInButton } from './internal/DummyCheckInButton/DummyCheckInButton'
+import { Debug } from './internal/Debug/Debug'
+import { DummyCamera } from './internal/DummyCamera'
+import { Typography } from '@material-ui/core'
+
+type CheckInType = 'event' | 'session'
+
+type Props = {
+  checkInType: CheckInType
+}
+
+export const CheckIn: React.FC<Props> = ({ checkInType }) => {
+  const [storedKeys, setStoredKeys] = React.useState<string[]>([])
+  const [open, setOpen] = useState(false)
+  useEffect(() => {
+    for (const key in localStorage) {
+      if (key.startsWith('check_in_')) {
+        setStoredKeys((prev) => (prev.includes(key) ? prev : [...prev, key]))
+      }
+    }
+
+    const interval = setInterval(() => {
+      console.log('Send check-in logs to dreamkast api:')
+      console.log(storedKeys)
+
+      for (const key of storedKeys) {
+        console.log(
+          `Success to send check-in log: key:L ${key}, value: ${JSON.stringify(
+            localStorage.getItem(key),
+          )}`,
+        )
+        deleteItem(key)
+      }
+    }, 10000)
```

Comment:
```suggestion
        localStorage.removeItem(key)
      }
      setStoredKeys([])
    }, 10000)
```

消すときも、stateの更新は一気にやったほうが良さそうです。
都度呼んでいることで、このuseEffectが複数回実行され、まだ削除されていないstoredKeys配列の後ろの方のkeyが複数回処理されていました（console.logの出力で確認しました）

Reply:
これ、都度呼んでいるだけでなく、storedKeysをfilterした値でreplaceせずにloopしているので、最後の要素しか消されないことに気付きました。
なので、例えばstoredKeysに4件入った状態で画面を読み込むと、storedKeysの最後しか消されないので、前半の方の要素は何度もAPI callされてしまっています。
backend側でat least onceを想定して冪等な実装にすればよいだけではありますが、できるだけexactly onceに近づけてあげたほうが良いと思います。

<img width="743" alt="image" src="https://github.com/cloudnativedaysjp/dreamkast-ui/assets/19190276/c6c095c1-ce86-4bcb-8c37-5b7289c5aaaa">